Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,422 @@
+            {
+                resultMessage
+                    .Append(FormatStringAndCommas(peopleList))
+                    .Append(peopleMessage)
+                    .Append(FormatCurrency(highestBalance, currencySymbol))
+                    .Append(messageEnd);
+                return resultMessage.ToString();
+            }
+            else
+            {
+                resultMessage.Clear();
+
+                resultMessage
+                    .Append(personWithHighestBalance)
+                    .Append(singlePersonMessage)
+                    .Append(FormatCurrency(highestBalance, currencySymbol))
+                    .Append(messageEnd);
+
+                return resultMessage.ToString();
+            }
+        }
+
+        /// <summary>
+        /// Return name and current money of the most poor person.
+        /// </summary>
+        public static string FindMostPoorPerson(string[] peopleAndBalances)
+        {
+            if (!ArrayIsValid(peopleAndBalances)) return invalidMessage;
+
+            const string peopleMessage = " have the least money. ";
+            const string singlePersonMessage = " has the least money. ";
+
+            var resultMessage = new StringBuilder();
+            var balanceList = new decimal[peopleAndBalances.Length];
+            var peopleList = new string[peopleAndBalances.Length];
+
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                balanceList[i] = CurrentBalanceForSinglePerson(peopleAndBalances[i]);
+                peopleList[i] = ReturnNameForSinglePerson(peopleAndBalances[i]);
+            }
+
+            var poorestPersonIndex = FindDecimalArrayMin(balanceList);
+            var poorestPerson = peopleList[poorestPersonIndex];
+            var poorestPersonMoney = balanceList[poorestPersonIndex];
+
+            if (balanceList.Length > 2 && ArrayElementsAreEqual(balanceList))
+            {
+                resultMessage
+                    .Append(FormatStringAndCommas(peopleList))
+                    .Append(peopleMessage)
+                    .Append(FormatCurrency(poorestPersonMoney, currencySymbol))
+                    .Append(messageEnd);
+                return resultMessage.ToString();
+            }
+
+            resultMessage.Clear();
+
+            resultMessage
+                .Append(poorestPerson)
+                .Append(singlePersonMessage)
+                .Append(FormatCurrency(poorestPersonMoney, currencySymbol))
+                .Append(messageEnd);
+
+            return resultMessage.ToString();
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (!ArrayIsValid(peopleAndBalances)) return invalidMessage;
+
+            //const string peopleMessage = " lost the most money. ";
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4970157)